library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -173,6 +233,72 @@ class Client() {
         }
     }
 
+    private suspend fun ffiXmtpClient(
+        options: ClientOptions?,
+        account: SigningKey?,
+        accountAddress: String,
+        appContext: Context?,
+        privateKeyBundleV1: PrivateKeyBundleV1,
+        legacyIdentitySource: LegacyIdentitySource,
+    ): FfiXmtpClient? {
+        val v3Client: FfiXmtpClient? =
+            if (options != null && options.enableLibXmtpV3 && options.appContext != null) {
```

Comment:
WDYT about having some validation where if you try and set `enableLibXmtpV3` and have the network set to an environment that doesn't support it we just error out of creation.

Reply:
For sure I can get that update to only work when on the local network.